flux-tests/tests/todo/list.rs

```diff
@@ -0,0 +1,67 @@
+#[flux::sig(fn(i32{v:false}) -> T)]
+pub fn never<T>(_x: i32) -> T {
+    loop {}
+}
+
+#[derive(Debug)]
+pub enum List<T> {
+    Nil,
+    Cons(T, Box<List<T>>),
+}
+
+pub fn len<T>(l: &List<T>) -> i32 {
+    match l {
+        List::Nil => 0,
+        List::Cons(_, tl) => 1 + len(&*tl),
```

Comment:
I think the following will also work, but I always get deref coercion wrong. Also, the auto-deref version may generate different code.

```suggestion
        List::Cons(_, tl) => 1 + len(tl),
```